### PR TITLE
chore: canonical keyboard version documentation

### LINKED ISSUE #19: [help.keyman.com] Hide deprecated keyboards from /keyboard index
From keymanapp/keyboards#252:

@LornaSIL writes:
> I would recommend that we remove deprecated keyboard help files from this page: http://https//help.keyman.com/keyboard/. The page is already a bit unwieldy and will get worse as we keep adding newer keyboards. The main (deprecated) keyboard page itself will still link to the deprecated help page.

@mcdurdin responds:
> To remove the deprecated items from the main index (or have them under a secondary index so that they can still be easily crawled if necessary), we'll need to do some work on help.keyman.com.

### LINKED ISSUE #824: chore: add rel=canonical and current-version links to keyboard documentation
Just as we have on the product documentation, we should have a similar pattern for keyboard documentation. This will help point search result at the latest version of documentation, addressing reports like: https://github.com/keymanapp/keyboards/pull/2395#issue-1937798912

### PR DESCRIPTION
**Pull request by mcdurdin**
* Fixes #824: adds a /keyboard/name/current-version canonical link
* Fixes #19: separates out deprecated keyboards and lists them later
* Also improves performance of keyboard index page, caching on first use.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -28,6 +28,13 @@ builder_describe \
 
 builder_parse "$@"
 
+function build_help_site() {
+  build_docker_container $HELP_IMAGE_NAME $HELP_CONTAINER_NAME
+  # if there is a cached keyboard index, remove it; it will be recreated on first use
+  # TODO: build this file during site build
```
Comment (darcywong00) [inline build.sh:34]: We prob need @tim-eves to help with this TODO. I'm guessing in [help-kubectl.yaml](https://github.com/keymanapp/help.keyman.com/blob/staging/resources/kubectl/help-kubectl.yaml)?

File: keyboard/index.php [hunk 2]
```diff
@@ -1,36 +1,49 @@
...
+        $items['deprecated'][$directory] = trim(substr($title, 0, $dep));
+      } else {
+        $items['active'][$directory] = $title;
+      }
     }
-    $items[$directory] = $title;
+
+    file_put_contents(INDEX_CACHE, json_encode($items));
+  } else {
+    $items = json_decode(file_get_contents(INDEX_CACHE), true);
```
Comment (darcywong00) [inline keyboard/index.php:44]: How does `INDEX_CACHE` get update when new keyboards are deployed (w/o rebuilding the site)

File: build.sh (hunk 1 above)
Comment (mcdurdin) [reply build.sh:34]: No, it goes right here in build.sh. It's low priority because it's built on first use anyway.

File: keyboard/index.php (hunk 2 above)
Comment (mcdurdin) [reply keyboard/index.php:44]: The site gets rebuilt when it's redeployed, doesn't it?

Comment (darcywong00) [reply keyboard/index.php:44]: ah right.

**Review by darcywong00 — APPROVED** [review]
lgtm

File: build.sh (hunk 1 above)
Comment (mcdurdin) [reply build.sh:34]: We'll move the cache build step to `configure` step, so it will be triggered by deploys in k8s

**Pull request merged**